examples/04_tool_inference_lock/ex_tool_inference_lock/vendor/inference_lock.rb

```diff
@@ -0,0 +1,147 @@
+# @see https://github.com/Eneroth3/inference-lock
+
+# Mixin for inference lock in tools.
+#
+# If any of the following Tool API methods are defined in the Tool class,
+# they must call `super` for this mixin to function:
+#
+# * `activate`
+# * `onKeyDown`
+# * `onKeyUp`
+# * `onKeyUp`
+# * `onMouseMove`
+#
+# `current_ip` must be implemented by the Tool class for this mixin to function.
+#
+# `start_ip` may be implemented for finer control.
+module Examples::ToolInferenceLock::InferenceLock
```

Comment:
I'm not convinced about this pattern about including modules that handles part of the Tool events, which the user then must remember to call `super` on. It also adds instance variables that isn't obvious from the point where you include (as a reader).

This feels more like it should be a class of its own that manages the state it needs isolated in its own instance. The tool would then manage all events and delegate to the class that manages the inference state. That ensures responsibilities are encapsulated and it's something you can actually write tests for.

Reply:
Agree. This can be made clearer and simpler.